bundles/amazon/ack/elasticache/bundle/config/01-replication-group.ytt.yml

```diff
@@ -0,0 +1,225 @@
+#@ else:
+      host: $(.replicationGroup.status.nodeGroups[0].primaryEndpoint.address)
+      port: $(.replicationGroup.status.nodeGroups[0].primaryEndpoint.port)
+#@ end
+    data:
+      password: "$(.creds.data.password)"
+---
+apiVersion: elasticache.services.k8s.aws/v1alpha1
+kind: User
+metadata:
+  name: #@ user_id
+  namespace: #@ data.values.namespace
+spec:
+  accessString: #@ users[user_name]["access_string"]
+  engine: #@ engine
+  passwords:
+  - name: #@ user_creds
+    key: password
+    namespace: #@ data.values.namespace
+  userID: #@ user_id
+  userName: #@ user_name
+  tags: #@ tags()
+#@ end
+---
+apiVersion: elasticache.services.k8s.aws/v1alpha1
+kind: UserGroup
+metadata:
+  name: #@ data.values.name
+  namespace: #@ data.values.namespace
+spec:
+  engine: #@ engine
+  userGroupID: #@ data.values.name
+  userIDs: #@ [ default_user["id"] ] + [ get_id(u) for u in users ]
+  tags: #@ tags()
+---
+apiVersion: elasticache.services.k8s.aws/v1alpha1
+kind: CacheParameterGroup
+metadata:
+  name: #@ cacheParameterGroupName
+  namespace: #@ data.values.namespace
+spec:
+  cacheParameterGroupFamily: #@ data.values.cacheParameterGroupFamily
+  cacheParameterGroupName: #@ cacheParameterGroupName
+  description: #@ "CacheParameterGroup for {} ReplicationGroup based on {} family".format(replicationGroupName, data.values.cacheParameterGroupFamily)
+  parameterNameValues:
+  - parameterName: cluster-enabled
+    parameterValue: #@ "yes" if data.values.clusterMode else "no"
+  tags: #@ tags()
+---
+apiVersion: elasticache.services.k8s.aws/v1alpha1
+kind: ReplicationGroup
+metadata:
+  name: #@ replicationGroupName
+  namespace: #@ data.values.namespace
+spec:
+  description: #@ "A {} service instance".format(engine)
+  engine: #@ engine
+  engineVersion: #@ data.values.engineVersion
+  replicationGroupID: #@ replicationGroupName
+  cacheNodeType: #@ data.values.cacheNodeType
+  cacheSubnetGroupName: #@ data.values.cacheSubnetGroupName
+  cacheParameterGroupName: #@ cacheParameterGroupName
+  securityGroupIDs: #@ data.values.vpcSecurityGroupIDs
+  atRestEncryptionEnabled: true
+  transitEncryptionEnabled: true
+#@ if/end data.values.clusterMode:
+  numNodeGroups: #@ data.values.numNodeGroups
+  replicasPerNodeGroup: #@ data.values.replicasPerNodeGroup
+  userGroupIDs:
+  - #@ data.values.name
+  tags: #@ tags()
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: #@ service_account_name
+  namespace: #@ data.values.namespace
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: #@ role_name
+  namespace: #@ data.values.namespace
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames: #@ [ users[u]["creds"] for u in users ]
+- apiGroups:
+  - elasticache.services.k8s.aws
+  resources:
+  - replicationgroups
+  verbs:
+  - get
+  - list
+  - watch
+  resourceNames:
+  - #@ data.values.name
```

Comment:
**nitpick** I know they are the same right now but using the var makes sense

```suggestion
  - #@ replicationGroupName
```

Reply:
👍